### ISSUE TITLE
Discrepency between `jax.jit` and `nnx.jit` for `static_argnames`

### ISSUE THREAD

**Issue by maxencefaldor** [google/flax#4146]
### System information
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): ubuntu:22.04
- Flax, jax, jaxlib versions (obtain with `pip show flax jax jaxlib`):
    - flax==0.8.5
    - jax==0.4.31
    - jaxlib==0.4.31
- Python version: 3.12.4
- GPU/TPU model and memory: NVIDIA RTX A6000
- CUDA version (if applicable): 12

### Problem you have encountered:
```
@partial(nnx.jit, static_argnames=("cond",))
def f(x, cond):
    if cond:
        return x
    else:
        return x + 1

f(jnp.array(1), True)
```
gives the error:
```
TracerBoolConversionError: Attempted boolean conversion of traced array with shape bool[].
The error occurred while tracing the function jit_fn at /src/conda/lib/python3.12/site-packages/flax/nnx/nnx/transforms/transforms.py:139 for jit. This concrete value was not available in Python because it depends on the value of the argument args[1].
See https://jax.readthedocs.io/en/latest/errors.html#jax.errors.TracerBoolConversionError
```

while doing the exact same thing with `jax.jit` instead of `nnx.jit` works:
```
@partial(jax.jit, static_argnames=("cond",))
def f(x, cond):
    if cond:
        return x
    else:
        return x + 1

f(jnp.array(1), True)
>>> Array(1, dtype=int32, weak_type=True)
```

The issue can be fixed by doing `f(jnp.array(1), cond=True)` instead of `f(jnp.array(1), True)`.

Is this behavior intended?

**Comment by cgarciae** [comment]
Seems to be fixed in version `0.9.0` we just released.

**Comment by maxencefaldor** [comment]
Indeed, thank you!
